io.home-assistant/makeBinarySensorDevices.py

```diff
@@ -0,0 +1,304 @@
+    "off": "normal",
+    "natural_on": ["hot", "too hot", "warm", "too warm"],
+  },
+  "light": {
+    "on": "detecting_light",
+    "off": "not_detecting_light",
+    "natural_on": ["detecting light", "detecting any light"],
+    "natural_off": ["not detecting light"],
+  },
+  "lock": {
+    "on": "unlocked",
+    "off": "locked",
+    "natural_on": ["unlocked", "open"],
+    "natural_off": ["locked", "closed"],
+  },
+  "moisture": {
+    "on": "wet",
+    "off": "dry",
+    "natural_on": ["wet", "damp", "moist"],
+  },
+  "motion": {
+    "on": "detecting_motion",
+    "off": "not_detecting_motion",
+    "natural_on": ["detecting motion", "detecting movement", "detecting any motion", "detecting any movement"],
+    "natural_off": ["not detecting motion"],
+  },
+  "moving": {
+    "on": "moving",
+    "off": "not_moving",
+    "natural_on": ["moving"],
+    "natural_off": ["not moving", "stationary"],
+  },
+  "occupancy": {
+    "on": "occupied",
+    "off": "not_occupied",
+    "natural_on": ["occupied"],
+    "natural_off": ["not occupied", "empty"],
+  },
+  "opening": {
+    "on": "open",
+    "off": "closed",
+  },
+  "plug": {
+    "on": "plugged",
+    "off": "unplugged",
+    "natural_on_token": ["plugged in"],
+  },
+  "power": {
+    "on": "detecting_power",
+    "off": "not_detecting_power",
+    "natural_on": ["detecting power", "detecting any power"],
+    "natural_off": ["not detecting power"],
+  },
+  "presence": {
+    "on": "home",
+    "off": "away",
+  },
+  "problem": {
+    "on": "detecting_problem",
+    "off": "not_detecting_problem",
+    "natural_on": ["detecting a problem", "detecting any problem"],
+    "natural_off": ["not detecting a problem"],
+  },
+  "safety": {
+    "on": "unsafe",
+    "off": "safe",
+  },
+  "smoke": {
+    "on": "detecting_smoke",
+    "off": "not_detecting_smoke",
+    "natural_on": ["detecting smoke", "detecting any smoke"],
+    "natural_off": ["not detecting smoke"],
+  },
+  "sound": {
+    "on": "detecting_sound",
+    "off": "not_detecting_sound",
+    "natural_on": ["detecting sound", "detecting any sound", "hearing anything", "hearing something"],
+    "natural_off": ["not detecting sound"],
+  },
+  "vibration": {
+    "on": "detecting_vibration",
+    "off": "not_detecting_vibration",
+    "natural_on": ["detecting vibration", "detecting any vibration", "vibrating", "shaking", "moving"],
+    "natural_off": ["not detecting vibration", "stationary"],
+  },
+  "window": {
+    "on": "open",
+    "off": "closed",
+  },
+}
+
+for device, states in supported_device_classes.items():
+
+  args = {
+    "device": device,
+    "device_name": ''.join([t.capitalize() for t in device.split('-')]),
+    "natural_on": [states["on"]],
+    "natural_off": [states["off"]],
+    "on": states["on"],
+    "off": states["off"],
+    "namespace": "io.home-assistant.binary-sensor.{}".format(device),
+  }
+
+  if not os.path.isdir(os.path.join(os.getcwd(), "../{namespace}".format(**args))):
```

Comment:
For future work you should check out `pathlib.Path`. It's amazing compared to hacking around with different `os` functions.